packages/mermaid/src/diagrams/common/svgDrawCommon.ts

```diff
@@ -1,89 +1,103 @@
-// @ts-nocheck - ignore to convert to TS
 import { sanitizeUrl } from '@braintree/sanitize-url';
+import type { Group, SVG } from '../../diagram-api/types.js';
+import type {
+  Bound,
+  D3ImageElement,
+  D3RectElement,
+  D3TSpanElement,
+  D3TextElement,
+  D3UseElement,
+  RectData,
+  TextData,
+  TextObject,
+} from './commonTypes.js';
 
-export const drawRect = function (elem, rectData) {
-  const rectElem = elem.append('rect');
-  rectElem.attr('x', rectData.x);
-  rectElem.attr('y', rectData.y);
-  rectElem.attr('fill', rectData.fill);
-  rectElem.attr('stroke', rectData.stroke);
-  rectElem.attr('width', rectData.width);
-  rectElem.attr('height', rectData.height);
-  rectElem.attr('rx', rectData.rx);
-  rectElem.attr('ry', rectData.ry);
+export const drawRect = (element: SVG | Group, rectData: RectData): D3RectElement => {
+  const rectElement: D3RectElement = element.append('rect');
+  rectData.x !== undefined && rectElement.attr('x', rectData.x);
+  rectData.y !== undefined && rectElement.attr('y', rectData.y);
+  rectData.fill !== undefined && rectElement.attr('fill', rectData.fill);
+  rectData.stroke !== undefined && rectElement.attr('stroke', rectData.stroke);
+  rectData.width !== undefined && rectElement.attr('width', rectData.width);
+  rectData.height !== undefined && rectElement.attr('height', rectData.height);
+  rectData.rx !== undefined && rectElement.attr('rx', rectData.rx);
+  rectData.ry !== undefined && rectElement.attr('ry', rectData.ry);
 
-  if (rectData.attrs !== 'undefined' && rectData.attrs !== null) {
+  if (rectData.attrs !== undefined && rectData.attrs !== null) {
     for (const attrKey in rectData.attrs) {
-      rectElem.attr(attrKey, rectData.attrs[attrKey]);
+      rectElement.attr(attrKey, rectData.attrs[attrKey]);
     }
   }
 
-  if (rectData.class !== 'undefined') {
-    rectElem.attr('class', rectData.class);
+  if (rectData.class !== undefined) {
```

Comment:
> Does changing `rectData.class !== 'undefined'` to `rectData.class !== undefined` do anything?
> 
> To me, it seems like this actually fixes a bug (if so, this should be in the PR description, so we can say something like "TypeScript caught some bad `undefined` handling, so this should now be fixed", then we can consider this PR to be both a bug fix, and a refactoring PR).

As far as I know, when using `typeof` before it would make it working as expected, but without it's comparing string, which means it has no value.

Reply:
Awesome :)

To be honest, we could even rename this PR to something like "Fix `undefined` handling in `svgDrawCommon`" if you want to have this as a bug-fix PR, instead of a refactoring PR!

Reply:
I would recommend leaving it as refactoring since the majority of the work is that; one line of change for a bug fix isn't showing the work done here. And that but fix it's has been highlighted in PR discerption.